rasa/core/policies/ensemble.py

```diff
@@ -526,14 +539,22 @@ def _pick_best_policy(
             if form_confidence > best_confidence:
                 best_policy_name = form_policy_name
 
-        return predictions[best_policy_name].probabilities, best_policy_name
+        best_prediction = predictions[best_policy_name]
+
+        # Apply policy events to tracker
+        policy_events += best_prediction.optional_events
+
+        for event in policy_events:
+            tracker.update(event, domain)
+
+        return best_prediction, best_policy_name
```

Comment:
why don't we put `best_policy_name` inside `best_prediction`?

Reply:
we'd loose the information about the index of the policy 😬

Reply:
as far as I know policy names aren't unique, right?

Reply:
aaahh. yes, we need an index. Maybe add a comment about it